Developer Guide Extension

SiHaoShen opened the pull request:
- Updated all the open source software (Grafana, InfluxDB, NodeRed)
- Changed NodeRed flows due to software update
- Add additional security aspect in docker-compose.yml: Linux capabilities, read-only file system, non-root user, container initialization

EndyCZ commented on src/solution/my_edge_app/docker-compose.yml:
```diff
@@ -33,27 +33,33 @@ services:
 
 ##### INFLUXDB ######
+  init-db:
+    image: init-db:v0.0.1               # Docker Init Container to Setup the Main InfluxDB v2.2 Container
```
Please add build statement with a path https://docs.docker.com/compose/compose-file/compose-file-v2/#build 
the goal is that user of the guide can just directly run docker-compose up -d commad after runnign the  [mosquite docker copose file](src/solution/mqtt_broker_mosquitto/docker-compose.yml)

EndyCZ commented on src/solution/my_edge_app/docker-compose_Edge.yml:
```diff
@@ -26,23 +26,32 @@ services:
+  
+  init-db:
+    image: init-db:v0.0.1               # Docker Init Container to Setup the Main InfluxDB v2.2 Container
+    mem_limit: 100m
```
Please add build statement with a path https://docs.docker.com/compose/compose-file/compose-file-v2/#build 
the goal is that user of the guide can just directly run docker-compose up -d commad after runnign the  [mosquite docker copose file](src/solution/mqtt_broker_mosquitto/docker-compose.yml)

EndyCZ commented on src/solution/my_edge_app/influxdb/init_influxdb.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -x
+set -e
+
+
```
Add comment that in this file we would load secrets from various Secrement management sources and use them to confugre the influx db database.

EndyCZ reviewed (changes requested):
I did not found any folder src/solution/tracee which you describes.

EndyCZ reviewed (changes requested):
Please also remove all the $http_proxy and $https_proxy. I think they are not needed anymore from the docker-compose files

SiHaoShen commented:
# Feedback from Review of 08.01
I removed all the $http_proxy and $https_proxys variable. Also I added build statement in **docker-compose.yml**, but not in **docker-compose_Edge.yml**. Since build statement is not supported in edge app publisher

EndyCZ commented:
Cannot import into IEAP image with enviroment variable. https://github.com/SiHaoShen/Developer-Guide-Hands-on-App/blob/main/src/solution/HandsOn_1/my_edge_app/docker-compose_Edge.yml#L29

EndyCZ reviewed (approved):
Merging now into dev branch to have it availalbe in organization repository. Could not merge because of Software clearing and some small details to adjust.
